ddtrace/internal/telemetry/writer.py

```diff
@@ -821,3 +828,47 @@ def _stop_service(self, join=True, *args, **kwargs):
         super(TelemetryWriter, self)._stop_service(*args, **kwargs)
         if join:
             self.join(timeout=2)
+
+    def _telemetry_excepthook(self, tp, value, root_traceback):
+        if root_traceback is not None:
+            # Get the frame which raised the exception
+            traceback = root_traceback
+            while traceback.tb_next:
+                traceback = traceback.tb_next
+
+            lineno = traceback.tb_frame.f_code.co_firstlineno
+            filename = traceback.tb_frame.f_code.co_filename
+            self.add_error(1, str(value), filename, lineno)
+
+            dir_parts = filename.split(os.path.sep)
+            # Check if exception was raised in the  `ddtrace.contrib` package
+            if "ddtrace" in dir_parts and "contrib" in dir_parts:
+                ddtrace_index = dir_parts.index("ddtrace")
+                contrib_index = dir_parts.index("contrib")
+                # Check if the filename has the following format:
+                # `../ddtrace/contrib/integration_name/..(subpath and/or file)...`
+                if ddtrace_index + 1 == contrib_index and len(dir_parts) - 2 > contrib_index:
+                    integration_name = dir_parts[contrib_index + 1]
+                    self.add_count_metric(
+                        "tracers",
+                        "integration_errors",
+                        1,
+                        (("integration_name", integration_name), ("error_type", tp.__name__)),
+                    )
+                    error_msg = "{}:{} {}".format(filename, lineno, str(value))
+                    self.add_integration(integration_name, True, error_msg=error_msg)
+
+            if self._enabled and not self.started:
+                self._app_started_event(False)
+
+            self.app_shutdown()
+
+        return self._ORIGINAL_EXCEPTHOOK(tp, value, root_traceback)
```

Comment:
This causes exceptions:
```
Traceback (most recent call last):
  File "/Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/ddtrace/internal/telemetry/writer.py", line 866, in _telemetry_excepthook
    return self._ORIGINAL_EXCEPTHOOK(tp, value, root_traceback)
TypeError: exceptiongroup_excepthook() takes 3 positional arguments but 4 were given
```

Reply:
You might have wanted something like
```
_ORIGINAL_EXCEPTHOOK = staticmethod(sys.excepthook)
```
in the class, but maybe it makes more sense to just stash `_ORIGINAL_EXCEPTHOOK` at the module level instead of as a class attribute.